pkg/networkservice/common/updatetoken/common.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updatetoken
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/token"
+)
+
+func updateToken(ctx context.Context, conn *networkservice.Connection, tokenGenerator token.GeneratorFunc) error {
+	// We need a real path if we don't have one
+	if conn == nil {
```

Comment:
Probably better to move this up into the Request case and simply create an empty connection if need be.

Reply:
Moved, but not sure it is better.